### PR TITLE
Add lr_scheduler options

### LINKED ISSUE #96: Add lr_scheduler options in BaseCV.configure_optimizers
The implementation of the configure_optimizer method of BaseCV apparently doesn't allow the use of a lr_scheduler.

This could be included in the 'optimizer' options maybe, for example, in a quick and dirty way

```
def configure_optimizers(self):
        """
        Initialize the optimizer based on self._optimizer_name and self.optimizer_kwargs.

        Returns
        -------
        torch.optim
            Torch optimizer
        """
        lr_scheduler_dict = self.optimizer_kwargs.pop('lr_scheduler', None)
    
        optimizer = getattr(torch.optim, self._optimizer_name)(
            self.parameters(), **self.optimizer_kwargs
        )
        if lr_scheduler_dict is not None:
            lr_scheduler_name = lr_scheduler_dict.pop('scheduler')
            lr_scheduler = {
                'scheduler': lr_scheduler_name(optimizer, **lr_scheduler_dict),
            }
            lr_scheduler_dict['scheduler'] = lr_scheduler_name
            self.optimizer_kwargs['lr_scheduler'] = lr_scheduler_dict
            return [optimizer] , [lr_scheduler]
        else: 
            return optimizer
```

### PR DESCRIPTION
**Pull request by EnricoTrizio**
## Description
Added the possibility to use torch learning rate scheduler.
They should be passed using the `options` in the CV models

This closes #96

## Status
- [x] Ready to go

### CODE REVIEW

File: mlcolvar/cvs/cv.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ def __init__(
         # OPTIM
         self._optimizer_name = "Adam"
         self.optimizer_kwargs = {}
+        self._lr_scheduler_name = None
```
Comment (andrrizzi) [inline mlcolvar/cvs/cv.py:46]: Do we need to cache this? Since ``self.lr_scheduler_kwargs`` is a public attribute, I'd keep it in there the whole time in case someone decide to change ``'scheduler'``.

File: mlcolvar/cvs/cv.py [hunk 2]
```diff
@@ -192,10 +196,18 @@ def configure_optimizers(self):
...
+
         optimizer = getattr(torch.optim, self._optimizer_name)(
             self.parameters(), **self.optimizer_kwargs
         )
-        return optimizer
+
+        if self.lr_scheduler_kwargs:
+            if self._lr_scheduler_name is None:
+                self._lr_scheduler_name = self.lr_scheduler_kwargs.pop('scheduler')
+            lr_scheduler = self._lr_scheduler_name(optimizer, **self.lr_scheduler_kwargs)
```
Comment (andrrizzi) [inline mlcolvar/cvs/cv.py:207]: If we remove ``self._lr_scheduler_name``, this would become something like
```python
scheduler_cls = self.lr_scheduler_kwargs['scheduler']
scheduler_kwargs = {k: v for k, v in self.lr_scheduler_kwargs.items() if k != 'scheduler'}
lr_scheduler = scheduler_cls(**scheduler_kwargs)
```

**Review by andrrizzi — APPROVED** [review]
Looks good but I'd suggest not to cache the ``self._lr_scheduler_name`` attribute internally.

File: mlcolvar/cvs/cv.py (hunk 1 above)
Comment (luigibonati) [reply mlcolvar/cvs/cv.py]: sounds good
